packages/testing/src/index.ts

```diff
@@ -9,35 +15,137 @@ import { ChildProcess, spawn, StdioOptions } from 'child_process';
 import events from 'events';
 import { kill, waitOnChild } from './child-process';
 import type getPortType from 'get-port';
+import { Connection, TestService } from './test-service-client';
+import { temporal } from '@temporalio/proto';
 
 const TEST_SERVER_EXECUTABLE_NAME = os.platform() === 'win32' ? 'test-server.exe' : 'test-server';
 
+export const DEFAULT_TEST_SERVER_PATH = path.join(__dirname, `../${TEST_SERVER_EXECUTABLE_NAME}`);
+
 /**
- * Options for {@link TestWorkflowEnvironment.create}
+ * Options passed to {@link WorkflowClient.result}, these are the same as the
+ * {@link BaseWorkflowResultOptions} with an additional option that controls
+ * whether to toggle time skipping in the Test server while waiting on a
+ * Workflow's result.
  */
-export interface TestWorkflowEnvironmentOptions {
+export interface WorkflowResultOptions extends BaseWorkflowResultOptions {
+  /**
+   * If set to `true`, waiting for the result does not unlock time skipping
+   */
+  runInNormalTime?: boolean;
+}
+
+/**
+ * Options passed to {@link WorkflowClient.execute}, these are the same as the
+ * {@link BaseWorkflowStartOptions} with an additional option that controls
+ * whether to toggle time skipping in the Test server while waiting on a
+ * Workflow's result.
+ */
+export type WorkflowStartOptions<T extends Workflow> = BaseWorkflowStartOptions<T> & {
+  /**
+   * If set to `true`, waiting for the result does not unlock time skipping
+   */
+  runInNormalTime?: boolean;
+};
+
+/**
+ * A client with the exact same API as the "normal" client with 1 exception,
+ * When this client waits on a Workflow's result, it will unlock time skipping
+ * in the test server.
```

Comment:
"unlock time skipping" is a weird phrase from a user's perspective. I'd just say "enable" or something.

Reply:
Yeah, the whole concept is confusing but it works well.
I'll try to clarify.